<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>prueba</title>
</head>

<body>
    <script>
        // this promise, is INSTANTLY REJECTED, because, the reject function is called before the resolve
        // for THAT, we should wrap the reject before, inside an if statement, and the
        // resolve inside an setTimeout, or another if statement.
        function makePizza(toppings = []) {
            const pizzaPromise = new Promise((resolve, reject) => {
                console.log(resolve, reject); // resolve and reject are functions
                reject(new Error('REJECTED!'));
                resolve(`here is your 🍕 with ${toppings.join(', ')}`);
            });
            return pizzaPromise;
        }
        const pizzaPepperoni = makePizza(['pepperoni']);
        console.log(pizzaPepperoni);

        // THE RESOLVE CAN CHECK A CONDITION.
        // THIS PROMISE IS ALLWAYS FULFILLED
        function wait(ms = 0) {
            return new Promise((resolve) => {
                setTimeout(() => {
                    resolve(ms === 200);
                }, ms);
            });
        }
        wait(2000).then((result) => {
            console.log(result); // FALSE. BUT FULFILLED
        });
        wait(200).then((result) => {
            console.log(result); // TRUE AND FULFILLED
        });
    </script>
</body>

</html>